include/zephyr/drivers/espi_emul.h

```diff
@@ -116,20 +53,61 @@ struct espi_emul {
 	/** Parent emulator */
 	const struct emul *parent;
 	/** API provided for this device */
-	const struct emul_espi_device_api *api;
+	const struct espi_emul_device_api *api;
 	/** eSPI chip-select of the emulated device */
 	uint16_t chipsel;
 };
 
-/** Definition of the eSPI controller emulator API */
-struct emul_espi_driver_api {
-	/* The struct espi_driver_api has to be first in
-	 * struct emul_espi_driver_api to make pointer casting working
-	 */
-	struct espi_driver_api espi_api;
-	/* The rest, emulator specific functions */
-	emul_trigger_event trigger_event;
-	emul_find_emul find_emul;
+/** Data about the virtual wire */
+struct espi_emul_vw_data {
+	/* Virtual wire signal */
+	enum espi_vwire_signal sig;
+	/* The level(state) of the virtual wire */
+	bool level;
+};
+
+struct espi_emul_device_api {
+	int (*config)(const struct espi_emul *dev,
+		struct espi_cfg *cfg);
+	bool (*get_channel_status)(const struct espi_emul *dev,
+		enum espi_channel ch);
+	/* Logical Channel 0 APIs */
+	int (*read_request)(const struct espi_emul *dev,
+		struct espi_request_packet *req);
+	int (*write_request)(const struct espi_emul *dev,
+		struct espi_request_packet *req);
+	int (*read_lpc_request)(const struct espi_emul *dev,
+		enum lpc_peripheral_opcode op, uint32_t *data);
+	int (*write_lpc_request)(const struct espi_emul *dev,
+		enum lpc_peripheral_opcode op, uint32_t *data);
+	int (*io_read)(const struct espi_emul *dev,
```

Comment:
What is the analog to this in espi_driver_api?

Reply:
Oh, ok... These don't have an analogical functions in real espi drivers. They are emulating the IO read/write through the eSPI bus which is handled internally by the EC chips. These require description.